app/scripts/controllers/transactions.js

```diff
@@ -153,13 +153,18 @@ module.exports = class TransactionController extends EventEmitter {
     const txParams = txMeta.txParams
     // ensure value
     txParams.value = txParams.value || '0x0'
-    this.query.gasPrice((err, gasPrice) => {
-      if (err) return cb(err)
-      // set gasPrice
-      txParams.gasPrice = gasPrice
-      // set gasLimit
-      this.txProviderUtils.analyzeGasUsage(txMeta, cb)
-    })
+    if (txParams.gasPrice) {
+      const hexString = txParams.gasPrice.toString(16)
```

Comment:
When was this a problem? Is `txParams.gasPrice` a BN? I thought it was already a hex, in which case `toString(16)` would only return a hex string because it already is one. It's possible all you really need to do is `addHexPrefix` here.

Maybe add a test to prove the issue?